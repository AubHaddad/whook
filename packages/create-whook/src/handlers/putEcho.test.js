import initPutEcho, { definition } from './putEcho';

describe('putEcho', () => {
  const log = jest.fn();

  it('should work', async () => {
    const putEcho = await initPutEcho({
      log,
    });
    const response = await putEcho({
      body: definition.operation.parameters[0].example,
    });

    expect({
      response,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });
});
