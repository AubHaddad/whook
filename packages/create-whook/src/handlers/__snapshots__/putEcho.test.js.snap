// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`putEcho should work 1`] = `
Object {
  "logCalls": Array [
    Array [
      "warning",
      "ðŸ“¢ - Echoing \\"Repeat this!\\"",
    ],
  ],
  "response": Object {
    "body": Object {
      "echo": "Repeat this!",
    },
    "status": 200,
  },
}
`;
